packages/ckb-sdk-utils/src/serialization/index.ts

```diff
@@ -14,58 +19,106 @@ const getOffsets = (elmLengths: number[]) => {
+ *           2. Serialize all items in it
+ */
+export const serializeFixVec = (fixVec: string | (string | Uint8Array)[]) => {
+  if (typeof fixVec !== 'string' && !Array.isArray(fixVec)) {
+    throw new TypeError('The fixed vector to be serialized should be a string or an array of bytes')
+  }
+  const vec = typeof fixVec === 'string' ? [...hexToBytes(fixVec)].map(b => `0x${b.toString(16)}`) : fixVec
+  const serializedArgVec = vec.map(serializeArray)
+  const header = toHexInLittleEndian(serializedArgVec.length)
+  return `${header}${serializedArgVec.join('')}`
 }
 
-export const serializeArgs = (args: string[]) => {
-  if (!args.length) return '04000000'
-  const serializedArgs = args.map(serializeArg)
-  const headerLength = fullLengthSize + offsetSize * serializedArgs.length
-  const fullLength = toHexInLittleEndian(headerLength + serializedArgs.join('').length / 2)
-  const offsets = getOffsets(serializedArgs.map(arg => arg.length / 2)).map(offset => toHexInLittleEndian(offset))
-  return [fullLength, ...offsets, ...serializedArgs].join('')
+/**
+ * @name serialzieDynVec
+ * @type DynVec, a dynamic-size vector is a vector whose items have a dynamic size
+ * @tutorial There are three steps of serializing a dynvec
+ *           1. Serialize the full size in bytes as a 32 bit unsigned integer in little-endian
+ *           2. Serialize all offset of the items as 32 bit unsigned integer in little-endian
+ *           3. Serialize all items in it
+ */
+export const serializeDynVec = (dynVec: (string | Uint8Array)[]) => {
+  if (!Array.isArray(dynVec)) {
+    throw new TypeError('The dynamic vector to be serialized should be an array of bytes')
+  }
+  const serializedArgVec = dynVec.map(arg => serializeArray(arg))
+  const body = serializedArgVec.join('')
+  let offsets = ''
+  if (serializedArgVec.length) {
+    offsets = getOffsets(serializedArgVec.map(serializedArg => serializedArg.length / 2))
+      .map(offset => toHexInLittleEndian(offset))
+      .join('')
+  }
+  const headerLength = fullLengthSize + offsetSize * serializedArgVec.length
+  const fullLength = toHexInLittleEndian(headerLength + body.length / 2)
+  return `${fullLength}${offsets}${body}`
 }
 
-export const serializeScript = (script: CKBComponents.Script) => {
-  if (!script) throw new Error('Script is required')
-  const { codeHash = '', hashType, args = [] } = script
-  const serializedCodeHash = serializeCodeHash(codeHash)
-  const serializedHashType = serializeHashType(hashType)
-  const serializedArgs = serializeArgs(args)
-  const elms = [serializedCodeHash, serializedHashType, serializedArgs]
-  const fullLength = toHexInLittleEndian(
-    fullLengthSize +
-      offsetSize * 3 +
-      (serializedCodeHash.length + serializedHashType.length + serializedArgs.length) / 2
-  )
-
-  const offsets = getOffsets(elms.map(elm => elm.length / 2)).map(offset => toHexInLittleEndian(offset))
-  return [fullLength, ...offsets, ...elms].join('')
+// TODO: add tests
```

Comment:
?

Reply:
There's only [one case](https://github.com/nervosnetwork/ckb-sdk-js/pull/342/files#diff-92f8e2032e0891a4819e0819fec1f14bR91) now
More tests should be added in the future